### PR TITLE
Add TIKVGroup controller syncing logic

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 

1. Add syncing controller for `TiKVGroup` and `TiDBGroup`
2. Add syncing manager for `TiKVGroup`
3. In tikvgroup syncing manager, tikvgroup will first register itself to the target tidbcluster.
4. tidbcluster manager will verify the registered groups each time.

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -237,6 +237,14 @@ type TidbClusterStatus struct {
...
 	// +optional
 	Conditions []TidbClusterCondition `json:"conditions,omitempty"`
+	// +optional
+	TiKVGroups []GroupRef `json:"tikv-groups,omitempty"`
+	// +optional
+	TiDBGroups []GroupRef `json:"tidb-groups,omitempty"`
+}
+
+type GroupRef struct {
+	Name string `json:"name"`
```
Comment (cofyc) [inline pkg/apis/pingcap/v1alpha1/types.go:247]: reuse v1.LocalObjectReference?

Comment (Yisaer) [reply pkg/apis/pingcap/v1alpha1/types.go]: updated.

File: pkg/manager/member/tikv_group_member_manager.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	klog.Infof("start to register tikvGroup[%s/%s] to tc[%s/%s]", tg.Namespace, tg.Name, tcNamespace, tcName)
+	tc, err := tgm.cli.PingcapV1alpha1().TidbClusters(tcNamespace).Get(tcName, metav1.GetOptions{})
+	if err != nil {
+		klog.Error(err)
+		return err
+	}
+
+	if tc.Status.TiKVGroups == nil || len(tc.Status.TiKVGroups) < 1 {
+		return tgm.registerTiKVGroup(tg, tc)
+	}
```
[the remark this answers is not in the document]
Comment (Yisaer) [reply pkg/manager/member/tikv_group_member_manager.go:68]: I don't know why we remove this block?

File: pkg/manager/member/tikv_group_member_manager.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+// checkWhetherRegistered will check whether the tikvgroup have already registered itself
+// to the target tidbcluster. If have already, the tikvgroup will be allowed to syncing.
+// If not, the tikvgroup will try to register itself to the tidbcluster and wait for the next round.
+func (tgm *tikvGroupMemberManager) checkWhetherRegistered(tg *v1alpha1.TiKVGroup) error {
+	tcName := tg.Spec.ClusterName
+	tcNamespace := tg.Namespace
+
+	klog.Infof("start to register tikvGroup[%s/%s] to tc[%s/%s]", tg.Namespace, tg.Name, tcNamespace, tcName)
+	tc, err := tgm.cli.PingcapV1alpha1().TidbClusters(tcNamespace).Get(tcName, metav1.GetOptions{})
```
[the remark this answers is not in the document]
Comment (Yisaer) [reply pkg/manager/member/tikv_group_member_manager.go]: updated.

File: pkg/manager/member/tikv_group_member_manager.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (DanielZhangQD) [reply pkg/manager/member/tikv_group_member_manager.go:67]: It seems L77 can cover the case of this block.

File: pkg/controller/tikvgroup/tikv_group_controller.go [hunk 4]
```diff
@@ -41,11 +47,22 @@ type Controller struct {
 func NewController(
 	kubeCli kubernetes.Interface,
 	cli versioned.Interface,
+	genericCli client.Client,
 	informerFactory informers.SharedInformerFactory,
 	kubeInformerFactory kubeinformers.SharedInformerFactory) *Controller {
+	eventBroadcaster := record.NewBroadcasterWithCorrelatorOptions(record.CorrelatorOptions{QPS: 1})
```
Comment (zjj2wry) [inline pkg/controller/tikvgroup/tikv_group_controller.go:53]: why QPS: 1? do not have default value?

File: pkg/controller/tikvgroup_control.go [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+	// don't wait due to limited number of clients, but backoff after the default number of steps
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		var updateErr error
+		updateTg, updateErr = rtc.cli.PingcapV1alpha1().TiKVGroups(ns).Update(tg)
+		if updateErr == nil {
+			klog.Infof("TiKVGroup: [%s/%s] updated successfully", ns, name)
+			return nil
+		}
+		klog.Errorf("failed to update TiKVGroup: [%s/%s], error: %v", ns, name, updateErr)
```
Comment (zjj2wry) [inline pkg/controller/tikvgroup_control.go:64]: should be warnning level?

Comment (Yisaer) [reply pkg/controller/tikvgroup_control.go]: updated.

File: pkg/controller/tikvgroup/tikv_group_controller.go (hunk 4 above)
Comment (Yisaer) [reply pkg/controller/tikvgroup/tikv_group_controller.go]: updated.

**Pull request merged**